### PR TITLE
Document that a non-PK column shouldn't be specified as the id-column [HZ-4211]

### PR DESCRIPTION
**Pull request by Edizc**
Added Microsoft SQL Server and Oracle to the supported databases and added a warning for declaring non-pk columns as the id-column.

### CODE REVIEW

File: docs/modules/mapstore/pages/configuring-a-generic-mapstore.adoc [hunk 1]
```diff
@@ -257,7 +257,7 @@ mapConfig.setMapStoreConfig(mapStoreConfig);
 ====
 
 |[[id-column]]`id-column`
-|Name of the column that contains the primary key.
+|Name of the column that contains the primary key. A column without a primary key constraint should not be specified as the id-column.
```
Comment (rebekah-lawrence) [inline docs/modules/mapstore/pages/configuring-a-generic-mapstore.adoc:260]:
```suggestion
|Name of the column that contains the primary key. Do not specify a column without a primary key constraint as the id-column.
```

Comment (rebekah-lawrence) [reply docs/modules/mapstore/pages/configuring-a-generic-mapstore.adoc:260]: Can we make the statement stronger, as suggested above, or is it only a recommendation (as implied by "should")?

**Review by rebekah-lawrence — COMMENTED** [review]
A suggestion with a related question

File: docs/modules/mapstore/pages/configuring-a-generic-mapstore.adoc (hunk 1 above)
Comment (Edizc) [reply docs/modules/mapstore/pages/configuring-a-generic-mapstore.adoc:260]: Yes it would be better like this.

**Review by rebekah-lawrence — APPROVED** [review]
LGTM

**Review by AleksPeychev — APPROVED** [review]

File: docs/modules/mapstore/pages/configuring-a-generic-mapstore.adoc [hunk 2]
```diff
@@ -257,7 +257,7 @@ mapConfig.setMapStoreConfig(mapStoreConfig);
 ====
 
 |[[id-column]]`id-column`
-|Name of the column that contains the primary key.
+|Name of the column that contains the primary key. Do not specify a column without a primary key constraint as the id-column.
```
Comment (frant-hartm) [inline docs/modules/mapstore/pages/configuring-a-generic-mapstore.adoc:260]: @Edizc does it really need to be primary key constraint? Wouldn't limiting to any unique key be enough?

**Review by frant-hartm — CHANGES_REQUESTED** [review]

File: docs/modules/mapstore/pages/configuring-a-generic-mapstore.adoc (hunk 2 above)
Comment (AleksPeychev) [reply docs/modules/mapstore/pages/configuring-a-generic-mapstore.adoc:260]: @frant-hartm you are right. I have tested it and limiting to any unique key is enough.

**Comment by AleksPeychev** [comment]
Fixes/Documents https://github.com/hazelcast/hazelcast/issues/26223

**Review by Serdaro — APPROVED** [review]

**Pull request merged**